lib/rjson/handler.rb

```diff
@@ -30,7 +32,17 @@ def push o
 
     def result
       root = @stack.first.last
-      process root.first, root.drop(1)
+      output = process root.first, root.drop(1)
+      if @truncated
+        if output.is_a? Hash
+          output["_truncated"] = true 
+        elsif output.is_a? Array
+          output.push "_truncated"
+        else
+          throw "JSON doc is a single, truncated scalar"
```

Comment:
Hmm - I was thinking that the valid json documents `"foo"` and `23` would hit this `else` block, and at that point we would want to throw an error, as those scalar values will probably be truncated, but won't have been flagged as `_truncated`.

However, the entire JSON parser crashes out on `"foo"` or `23` with an error as follows, so I think I'll just remove this for now.
```
[3] pry(main)> RJSON.load '"foo"'
NoMethodError: undefined method `first' for :root:Symbol
from /Users/samjewell/bu/rjson/lib/rjson/handler.rb:35:in `result'
[4] pry(main)> RJSON.load '12'
NoMethodError: undefined method `first' for :root:Symbol
from /Users/samjewell/bu/rjson/lib/rjson/handler.rb:35:in `result'
```